### PR TITLE
gha: Update kube-proxy-replacement flag values

### PR DESCRIPTION
**Pull request by sayboras**
Please refer to individual commits for more details.

### CODE REVIEW

**Review by bimmlerd — APPROVED** [review]

File: .github/workflows/conformance-k8s-network-policies.yaml [hunk 1]
```diff
@@ -93,7 +93,7 @@ jobs:
              --wait \
              --namespace kube-system \
              --set nodeinit.enabled=true \
-             --set kubeProxyReplacement=partial \
+             --set kubeProxyReplacement=true \
              --set socketLB.enabled=false \
              --set externalIPs.enabled=true \
              --set nodePort.enabled=true \
```
Comment (julianwiedmann) [inline .github/workflows/conformance-k8s-network-policies.yaml]: I think you want `=false` here, and let all the specific feature options take effect.

**Review by julianwiedmann — DISMISSED** [review]

File: .github/workflows/conformance-k8s-network-policies.yaml (hunk 1 above)
Comment (sayboras) [reply .github/workflows/conformance-k8s-network-policies.yaml]: ah good spot :bow:

**Review by julianwiedmann — APPROVED** [review]
lgtm, thank you Tam! :broom:

File: .github/workflows/conformance-externalworkloads.yaml [hunk 2]
```diff
@@ -149,7 +149,7 @@ jobs:
           CILIUM_INSTALL_DEFAULTS="${{ steps.default_vars.outputs.cilium_install_defaults }} \
             --cluster-name=${{ env.clusterName }} \
             --datapath-mode=tunnel \
-            --helm-set kubeProxyReplacement=strict"
+            --helm-set kubeProxyReplacement=true"
```
Comment (ghost) [inline .github/workflows/conformance-externalworkloads.yaml:152]: In some places this option is set with `--helm-set-string`, but here and in `conformance-multi-pool.yaml` with just `--helm-set`. I vaguely recall that the string type is recognized automatically? Just wanted to double check.

**Review by ghost — APPROVED** [review]
docs good

**Review by thorn3r — APPROVED** [review]
LGTM :+1:

File: .github/workflows/conformance-externalworkloads.yaml (hunk 2 above)
Comment (sayboras) [reply .github/workflows/conformance-externalworkloads.yaml:152]: I am under the same impression, considered that CI is green, I think it should be fine.

Will keep an eye on this point still.

**Review by tklauser — APPROVED** [review]

**Pull request merged**